src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -384,6 +396,14 @@ string GetHeaderSeparator (string name)
 		async Task<NSUrlRequest> CreateRequest (HttpRequestMessage request)
 		{
 			var stream = Stream.Null;
+			// set header cookies if needed from the managed cookie container
+			CookieCollection cookies = cookieContainer?.GetCookies (request.RequestUri);
+			if (cookies != null && cookies.Count > 0) {
+				foreach (var cookie in cookies) {
+					Console.WriteLine ($"Adding cookie: {cookie.ToString ()} ");
+					request.Headers.TryAddWithoutValidation (Cookie, cookie.ToString ()); // as per docs: Returns a string representation of this Cookie object that is suitable for including in a HTTP Cookie: request header.
```

Comment:
Not sure the standard allows having multiple Cookie headers.
I think the recommendation is to "join" all cookies into a single Cookie header using the ; character.

https://stackoverflow.com/questions/16305814/are-multiple-cookie-headers-allowed-in-an-http-request

Reply:
We might want to use `GetCookieHeader` instead of `GetCookies`.

It looks that this is the approach that [AndroidClientHandler.cs](https://github.com/xamarin/xamarin-android/blob/master/src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs#L853) takes.

Reply:
It is a very good point. And I won't have to deal with the specific Header details!